project_event/models/project_task.py

```diff
@@ -282,12 +282,43 @@ def create_new_activity(self, vals):
             .next_by_code('project.task.activity')
         return super(Task, self).create(vals)
 
+    @api.multi
+    def update_reservation_event(self, update_task):
+        if update_task and self.reservation_event_id:
```

Comment:
self.ensur_one at the beginning. rather than using update_task, you can use self